cli/module_registry.go

```diff
@@ -467,6 +484,60 @@ func validateModuleFile(client *viamClient, moduleID moduleID, tarballPath, vers
 		entrypoint, extraErrInfo)
 }
 
+func validateDynamicExecutableLinkedLibaries(client *viamClient, tarballPath, entrypoint, platform string) error {
+	if runtime.GOOS != "linux" || !strings.HasPrefix(platform, "linux") {
+		return nil
+	}
+	tempDir := os.TempDir()
+	//nolint:errcheck
+	defer os.Remove(tempDir)
+	err := unpackArchive(tarballPath, tempDir)
+	if err != nil {
+		return errors.Wrapf(err, "failed to unpack archive for validation")
+	}
+	//nolint:gosec
+	cmd := exec.Command("ldd", filepath.Join(tempDir, entrypoint))
+	out, err := cmd.Output()
+	if err != nil {
+		//nolint:errorlint
+		exitErr, ok := err.(*exec.ExitError)
+		if !ok {
+			return errors.Wrapf(err, "err from exec.Command was not an ExitError")
+		}
+		// this is fine -- the entrypoint is likely a script or static binary
+		if strings.Contains(string(exitErr.Stderr), "not a dynamic executable") {
+			return nil
+		}
+		return errors.Wrapf(err, "failed to run ldd on the module entrypoint")
+	}
+	expectedSubstrings := []string{
+		"ld-linux",
+		"glibc",
+		"linux-vdso",
+		tempDir,
+	}
+	lines := strings.Split(string(out), "\n")
+	for _, line := range lines {
+		line := strings.TrimSpace(line)
+		if line == "" {
+			continue
+		}
+		foundLegalSubstr := false
+		for _, expectedSubstr := range expectedSubstrings {
+			if strings.Contains(line, expectedSubstr) {
+				foundLegalSubstr = true
+				break
+			}
+		}
+		if !foundLegalSubstr {
+			warningf(client.c.App.ErrWriter, "The module's entrypoint is a dynamic executable which depends on a library "+
```

Comment:
discussed with abe offline. Will coalesce all warnings into a single larger message. Otherwise it will be too much clutter